test/suites/drift/suite_test.go

```diff
@@ -476,6 +476,59 @@ var _ = Describe("Drift", func() {
 		Entry("MaxPods", v1beta1.AKSNodeClassSpec{MaxPods: lo.ToPtr[int32](10)}),
 	)
 
+	Context("FIPS Drift", func() {
+		BeforeEach(func() {
+			if env.InClusterController {
+				Skip("FIPS drift tests require SIG access - skipping in self-hosted mode")
+			}
+		})
+
+		DescribeTable("AKSNodeClass FIPS", func(initialNodeClassSpec, updatedNodeClassSpec v1beta1.AKSNodeClassSpec) {
+			// Apply initial modifications to ensure we start with the right base state
+			initialNodeClass := test.AKSNodeClass(v1beta1.AKSNodeClass{Spec: *nodeClass.Spec.DeepCopy()}, v1beta1.AKSNodeClass{Spec: initialNodeClassSpec})
+			initialNodeClass.ObjectMeta = nodeClass.ObjectMeta
+
+			// Create the updated nodeClass for drift
+			updatedNodeClass := test.AKSNodeClass(v1beta1.AKSNodeClass{Spec: *initialNodeClass.Spec.DeepCopy()}, v1beta1.AKSNodeClass{Spec: updatedNodeClassSpec})
+			updatedNodeClass.ObjectMeta = nodeClass.ObjectMeta
+
+			env.ExpectCreated(dep, initialNodeClass, nodePool)
+
+			nodeClaim := env.EventuallyExpectRegisteredNodeClaimCount("==", 1)[0]
+			pod := env.EventuallyExpectHealthyPodCount(selector, numPods)[0]
+			node := env.ExpectCreatedNodeCount("==", 1)[0]
```

Comment:
can we move pod to the front here? 

If nodeclaim is registered but not initialized we may run into problems where we replace that nodeclaim, and we get not found errors that cause flake.